### PR TITLE
Refactor to v1beta2

### PR DESCRIPTION
**Pull request by Juanito98**
Refactor to suppport v1beta2 endpoints:

* chat/completions
* completions
* embeddings

This pkg usage will follow the common industry implementation.

Now a chat completion would look like:

```typescript
import * as replitai from '@replit/ai-modelfarm';
const result = await replitai.chat.completions.create({
  model: 'chat-bison',
  messages: [
    {
      role: 'user',
      content: 'What is the meaning of life?',
    },
  ],
  temperature: 0.5,
  max_tokens: 128,
});
```

## Testplan

* Modify tests for new modelfarm client
* Test passes

### CODE REVIEW

**Comment by sergeichestakov** [comment]
this looks good to me but don't have all the context here will let Faris / James tal

**Review by masad-frost — COMMENTED** [review]
I don't understand why we should go with a class based approach over using modules. I don't know that there would be any state that we need to build up within those classes. The URL configuration seems useful for testing and development, but the env var seems to be enough, and if we really have to we can drill that option with every call.

OpenAI TypeScript API isn't something that people love, quite the opposite, it's extremely clunky and non idiomatic, the type errors it generated is also bad. I'm not sure I agree that we should follow their API design (or lack thereof since the library was autogenerated).

If we're conforming the API to match OpenAI's, I wonder if we just get rid of this library, or at most keep it as a thin wrapper around OpenAI's library.

**Comment by masad-frost** [comment]
Coincidentally, just came across this from a prominent AI and JavaScript community member (blurred name since this is their private Twitter) 
Screenshot_20231221-181316

**Comment by Juanito98** [comment]
Hey @masad-frost
I liked the module approach and error handling. Addressed now.
Can I have another 👁️ ?

File: src/request/handleStreamingResponseBody.ts [hunk 1]
```diff
@@ -24,7 +23,7 @@ export default function handleStreamingResponseBody<T, R>(
 
         for (const json of parserSource) {
           // Ideally we do some assertions on T here
-          yield processJSON(json as T);
```
[the remark this answers is not in the document]
Comment (Juanito98) [reply src/request/handleStreamingResponseBody.ts:27]: Do you think it's worth it to add some zod / typebox for validation?

**Review by masad-frost — APPROVED** [review]
Code looks really good thanks for the PR. Left some comments.

I think I have a different view philosophically of how model farm should look, but I'll trust that you and @JimmyAustin discussed this and have some vision here, I'll get the scoop offline.

File: src/index.ts [hunk 2]
```diff
@@ -1,46 +1,10 @@
...
-  CompletionOptions,
-  CompletionMultipleChoicesOptions,
-  ChatMessage,
-  CompletionModel,
-  ChatModel,
-  RequestError,
-  Embedding,
-  EmbeddingModel,
-  EmbeddingOptions,
-};
```
[the remark this answers is not in the document]
Comment (Juanito98) [reply src/index.ts:46]: Exported root functions, thanks

**Pull request merged**